Complete shop svs

Sean-Stille opened the pull request:
Admin side of things should be complete now. An admin will select which item a group will be used for and will only see items that are meant for the groups they are part of. 

From here the member side just needs to be done, which should be pretty quick, since it'll likely use the same database

Wil-LaLonde commented on AcademicReward/Logic/ShopLogic.cs:
```diff
@@ -42,9 +42,11 @@ public LogicErrorType AddItem(object obj)
-            
-            
+            toBeAdded = new ShopItem(AddItemVals[0], AddItemVals[1], cost, level, 
+                MauiProgram.Profile.GroupList.ElementAt( int.Parse(AddItemVals[5]) ) );
```
You could always do a TryParse here, that way if anything were to go wrong, the whole application wouldn't crash. Not sure how it would go wrong, but you never know

Sean-Stille commented:
I've added in the member views of the shop, and implemented their ability to buy an item, currently I don't think the point cost is applied to the account. Additionally, purchasing an item doesn't seem to remove it from the shop currently

Sean-Stille commented:
Now when an admin creates a shop item, it adds to the history table

Wil-LaLonde reviewed (commented):
Everything looks good here, just take a look at my one comment regarding the int.Parse.

Otherwise, in order to decrement the points, it appears that you are decrementing them but only locally. You'll have to make a database call to update the point value for the given ProfileID. I'm not sure why the shop item wouldn't be removing after buying it. Does it need to be inserted in the PurchasedItem table?

pattmax00 commented:
> I'm not sure why the shop item wouldn't be removing after buying it. Does it need to be inserted in the PurchasedItem table?

It might have something to do with the pivot table containing its primary key as a reference, also see my comment on Discord, I'm not sure that we want to be removing items from the shop after they are purchased.
